### ISSUE TITLE
FRAGE: können auch Aktennr. und Betreff über den externen launcher übergeben werden? 

### ISSUE THREAD

**Issue by iradraconis** [jlawyerorg/j-lawyer-org#2601]
Ich bin durch eine Frage im Forum darauf gestoßen und habe das einfach mal ausprobiert. Mit der folgenden Konfiguration können PDF-Dateien per Thunderbird verschickt werden. Die Dateien werden in den Anhang gelegt, allerdings bleiben Betreff und alles andere leer.

Geht das schon oder wäre das ein neues Feature? 
In der Dokumentation habe ich nichts dazu gefunden. Die Parameter sind auch nicht erläutert.

![2024-10-05 14 14 55](https://github.com/user-attachments/assets/bca86f3c-9fd4-48fb-92c1-23b0bc9ac344)

**Comment by iradraconis** [comment]
Wenn das nicht geht, Frage an @j-dimension : hat das Kontextmenü noch Platz für einen weiteren Eintrag? Bzw. wärst du dafür zu gewinnne? 

Dann könnte ich mich daran versuchen, Thunderbird als Versandoption verfügbar machen. Da wir ohnehin auch eine Erweiterung für Thunderbird haben, wäre das ein nützliches Feature

**Comment by iradraconis** [comment]
Alternative zum Kontextmenü als Idee: 
im eMail Dialog könnte ein Button hinzugefügt werden, der alle Kontextdaten übernimmt und eine neue E-Mail in Thunderbird öffnet.

**Comment by j-dimension** [comment]
Es kann dort ausschließlich der Dateiname übergeben werden. Alles weitere wäre eine Erweiterung.

**Comment by j-dimension** [comment]
> hat das Kontextmenü noch Platz für einen weiteren Eintrag? Bzw. wärst du dafür zu gewinnne?

Das machen wir dann schon irgendwie passend.

**Comment by iradraconis** [comment]
> > Mit der folgenden Konfiguration können PDF-Dateien per Thunderbird verschickt werden.
> 
> Bist Du sicher, dass dann auch der korrekte Inhalt in der Anlage enthalten ist?

Ja funktioniert bei mir (MacOS).. Sollte das nicht funktionieren?

**Comment by j-dimension** [comment]
> Ja funktioniert bei mir (MacOS).. Sollte das nicht funktionieren?

Nein, alles bestens, der Platzhalter verweist ja bereits auf die temporäre Datei. Das wird sicher funktionieren.

**Comment by iradraconis** [comment]
Können auch Java Applikationen (lauffähige jar Dateien) mit dem Launcher gestartet werden, einschließlich übergebener Parameter?

**Comment by j-dimension** [comment]
Ja, das wird funktionieren. Es gibt aber letztlich nur den DATEINAME-Parameter, der dynamisch gebildet wird. Feste Parameter kann man aber in beliebiger Menge übergeben.

**Comment by iradraconis** [comment]
Wie würde die Konfiguration denn aussehen? Als Programm dann die jar Datei auswählen? Welche Bedeutung haben die möglichen Parameter, w P etc. ?

**Comment by j-dimension** [comment]
Grundsätzlich wird ein .jar wie folgt aufgerufen:

`java -jar dings.jar`

Dazu müsste ein Java im PATH zu finden sein. Alternativ den vollen Pfad zum java-Executable angeben. Es liegt eines im Unterverzeichnis des Clients. 

Was meinst Du mit w und P - wo nimmst Du die her?

**Comment by iradraconis** [comment]
> Grundsätzlich wird ein .jar wie folgt aufgerufen:
> 
> `java -jar dings.jar`
> 

Das weiß ich :) Ich habe zu kompliziert gedacht. Habe es ans Laufen bekommen.


> Was meinst Du mit w und P - wo nimmst Du die her?

Daher:
![2024-10-09 21 46 53](https://github.com/user-attachments/assets/ee5f2340-427a-4274-b685-dfb4f103452c)

**Comment by j-dimension** [comment]
Das ist Mac-spezifisch für den Standard-Opener /usr/bin/open. Nach -a erscheint der Name der zu startenden Applikation.

-W lässt den Prozess "blockieren". Er wird also nicht im Hintergrund / asynchron gestartet, sondern das aufrufende Programm wird erst weiter ausgeführt, wenn die externe Applikation beendet wurde. Das gibt dem Aufrufer mehr Kontrolle.

**Comment by iradraconis** [comment]
Das müsste in die Dokumentation :) Danke. Hab mich schon immer gefragt, was die Parameter machen.
